python_modules/dagster/dagster/_core/definitions/time_window_partition_mapping.py

```diff
@@ -207,6 +223,45 @@ def _map_partitions(
 
                 time_windows.append(TimeWindow(window_start, window_end))
 
+        current_time = cast(
+            datetime,
+            pendulum.instance(current_time, tz=to_partitions_def.timezone)
+            if current_time
+            else pendulum.now(to_partitions_def.timezone),
+        )
+
+        first_window = to_partitions_def.get_first_partition_window(current_time=current_time)
+        last_window = to_partitions_def.get_last_partition_window(current_time=current_time)
+        if raise_on_non_existent_partition:
```

Comment:
also a nitpick, but could we save some lines of code / logic by just generating the filtered set of time windows regardless of this flag, then only raising the error if this flag is set and the filtered time windows don't equal the non-filtered time windows?

Reply:
makes sense!